libarchive/archive_write.c

```diff
@@ -354,6 +354,10 @@ archive_write_client_open(struct archive_write_filter *f)
 	int ret;
 
 	f->bytes_per_block = archive_write_get_bytes_per_block(f->archive);
```

Comment:
I see what you're trying to protect against and I think there's a simpler fix:

Change `archive_write_set_bytes_per_block` so that it will never set a value less than zero.  (Ignore the provided value if it's less than zero.)   With that change, I think we can then guarantee that `get_bytes_per_block` will never return a negative value.